src/clr-angular/forms/datepicker/date-container.ts

```diff
@@ -140,6 +161,19 @@ export class ClrDateContainer implements DynamicWrapper, OnDestroy {
         this.invalid = invalid;
       })
     );
+
+    /** handle when to show are hide the clear button */
+    this.subscriptions.push(
+      this.dateIOService.value.subscribe((value: string) => {
+        this.inputValue = value;
```

Comment:
Maybe you already tried this and it didn't work. But, what happens if you subscribe in `ngAfverViewInit`? That seems like the place to subscribe because the value isn't set until that lifecycle hook in date-input.ts